operator/main.go

```diff
@@ -148,6 +159,79 @@ func runOperator(cmd *cobra.Command) {
 	}
 	close(k8sInitDone)
 
+	k8sversion.Update(k8s.Client(), option.Config)
+	if !k8sversion.Capabilities().MinimalVersionMet {
+		log.Fatalf("Minimal kubernetes version not met: %s < %s",
+			k8sversion.Version(), k8sversion.MinimalVersionConstraint)
+	}
+
+	// Get hostname for identity name of the lease lock holder.
+	// We identify the leader of the operator cluster using hostname.
+	hostID, err := os.Hostname()
+	if err != nil {
+		log.WithError(err).Fatal("failed to get hostname")
+	}
+	hostID = rand.RandomStringWithPrefix(hostID+"-", 8)
+	log.Infof("using host ID %s for operator POD", hostID)
+
+	ns := option.K8sNamespaceName
+	// If due to any reason the NAMESPACE is not set we assume it to be
+	// in default namespace.
+	if ns == "" {
+		ns = metav1.NamespaceDefault
+	}
+
+	// Use a Go context so we can tell the leaderelection code when we
+	// want to step down
+	ctx, contextCancelFunc := context.WithCancel(context.Background())
+
+	leaseLock := &resourcelock.LeaseLock{
+		LeaseMeta: metav1.ObjectMeta{
+			Name:      leaderElectionLeaseLockName,
+			Namespace: ns,
+		},
+		Client: k8s.Client().CoordinationV1(),
+		LockConfig: resourcelock.ResourceLockConfig{
+			// Identity name of theholder
+			Identity: hostID,
+		},
+	}
+
+	// Start the leader election for running cilium-operators
+	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
+		Name: leaderElectionLeaseLockName,
+
+		Lock:            leaseLock,
+		ReleaseOnCancel: true,
+
+		LeaseDuration: operatorOption.Config.LeaderElectionLeaseDuration,
+		RenewDeadline: operatorOption.Config.LeaderElectionRenewDeadline,
+		RetryPeriod:   operatorOption.Config.LeaderElectionRetryPeriod,
+
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: onOperatorStartLeading,
+			OnStoppedLeading: func() {
+				log.Infof("leader election lost: %s", hostID)
```

Comment:
nits: do you want to keep it consistent for logging, something like log.WithField("lease_id", hostID).Info("leader election lost") ?

Reply:
Yeah, I think it makes sense. I will update the logging statements.